modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/hashed/files/osgi/util/tracker/HashedFileURIsRegistryServiceTrackerCustomizer.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.frontend.js.web.internal.hashed.files.osgi.util.tracker;
+
+import com.liferay.frontend.js.web.internal.hashed.files.HashedFileURIsRegistry;
+import com.liferay.petra.string.StringPool;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.servlet.ServletContext;
+
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
+import org.osgi.util.tracker.ServiceTrackerCustomizer;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class HashedFileURIsRegistryServiceTrackerCustomizer
+	implements ServiceTrackerCustomizer<ServletContext, Map<String, String>> {
+
+	public HashedFileURIsRegistryServiceTrackerCustomizer(
+		BundleContext bundleContext,
+		HashedFileURIsRegistry hashedFileURIsRegistry) {
+
+		_bundleContext = bundleContext;
+		_hashedFileURIsRegistry = hashedFileURIsRegistry;
+	}
+
+	@Override
+	public Map<String, String> addingService(
+		ServiceReference<ServletContext> serviceReference) {
+
+		ServletContext servletContext = _bundleContext.getService(
+			serviceReference);
+
+		try {
+			String contextPath = servletContext.getContextPath();
+
+			Set<String> hashedResourcePaths = _getHashedResourcePaths(
+				servletContext, "/META-INF/resources/__liferay__/");
+
+			Map<String, String> map = new HashMap<>();
+
+			for (String hashedResourcePath : hashedResourcePaths) {
+
+				// Remove "/META-INF/resources" from path
+
+				hashedResourcePath = hashedResourcePath.substring(19);
```

Comment:
This is my java ignorance here, but can we do something similar to `replace("/META-INF/resources", "")`? Primarily just thinking for find/replace in the future

Reply:
You can, but it's slower because you need to traverse the string and all that, whereas a `substring(19)` is just a memory indexing operation.